### ISSUE TITLE
Programmatically way to set parameters

### ISSUE THREAD

**Issue by ferchaure** [jniediek/combinato#48]
Hi Johannes,

Is there a way to edit all the parameters programmatically?
I read that I can make my own local_options.py.  But I saw other important parameters on _combinato/combinato/extract/extract_spikes.py_ is it possible to have them on local_option.py ?

If you think that is ok, I can code a pull request where extract_spikes.py try to load its parameters from local_option.py if they are there.

And maybe the same for:  _combinato/combinato/basics/filters.py_  that has a few more parameters.

**Comment by jniediek** [comment]
There is no general way to change those parameters that are defined on a per-file basis, such as the ones in `extract_spikes.py`. If you decide to add those options to `local_options.py`, the syntax for loading could be the following:

```
try:
    from local_options import options as local_options
    options.update(local_options)
    print('Updated {} options by local_options'.format(len(local_options)))
except ImportError:
    pass
```

**Comment by ferchaure** [comment]
That do you think about counting the updated parameters:

```
try:
    from local_options import options as local_options
    options.update(local_options)
    print('Updated {} options by local_options'.format(len(set(local_options.keys()) & set(options.keys()))))
except ImportError:
    pass
```

another question: is it there a number version of combinato inside the code?
Spikeinterface has a method asking for that (probably to reinforce reproducibility).

**Comment by ferchaure** [comment]
Just to check,  combinato analyze each channel of a polytrode independently?

**Comment by jniediek** [comment]
> That do you think about counting the updated parameters:
> 
> ```
> try:
>     from local_options import options as local_options
>     options.update(local_options)
>     print('Updated {} options by local_options'.format(len(set(local_options.keys()) & set(options.keys()))))
> except ImportError:
>     pass
> ```

In principle I agree to count. But your syntax doesn't really count _updating_, either. Imagine there are keys in `local_options` and `options` whose values are identical. So to count precisely, we would need to check for each key that exists in `local_options` and `options`, whether the value differs.

Moreover, if there are keys in `local_options` that don't exist in `options`, this could also be called updating, because `options` are updated to what we want them to be.

So my suggestion is to leave the code as is, even if it's not 100% precise in this point.

**Comment by jniediek** [comment]
> another question: is it there a version number of combinato inside the code?
> Spikeinterface has a method asking for that (probably to reinforce reproducibility).

No, there is no version number. One could use [gitpython](https://gitpython.readthedocs.io/en/stable/intro.html) to extract the current commit ID, but this seems like overkill to me at the moment. What do you suggest?

**Comment by jniediek** [comment]
> Just to check, combinato analyze each channel of a polytrode independently?

Combinato does not have explicit polytrode support. I did some tests with concatenated data (for each spike, I concatenated all polytrode traces one after the other into one artificial "spike"). This worked well, but it's not the same as explicit support for polytrodes.

**Comment by ferchaure** [comment]
> No, there is no version number. One could use gitpython to extract the current commit ID, but this seems like overkill to me at the moment. What do you suggest?

Maybe a txt file with a number like 1.0.1 that just increase for changes in the core algorithm. I did that for wave_clus, it gives a friendly way to report which code you used in a paper .

> Combinato does not have explicit polytrode support. I did some tests with concatenated data (for each spike, I concatenated all polytrode traces one after the other into one artificial "spike"). This worked well, but it's not the same as explicit support for polytrodes.

Ok, I know exactly what you mean.

>Moreover, if there are keys in local_options that don't exist in options, this could also be called updating, because options are updated to what we want them to be.
So my suggestion is to leave the code as is, even if it's not 100% precise in this point.

Ok

**Comment by ferchaure** [comment]
Hi again. I'm having issues importing  local_options.py. I understood that that file should be in the same folder that the data, and from there the _css-_ commands should be executed. But that folder (the current directory for python) is not included in the sys.path to do the imports. At least in windows, this could be fixed adding in combinato\__init__.py the lines:
```
import os,sys
sys.path.insert(0, os.getcwd())
```
Is this ok for you? What do you think? I will check on Linux in a few hours

**Comment by ferchaure** [comment]
I tested it on linux and it is working; at least with that change on combinato\\\__init__.py 

The interface works in this way:

**Setup**
1. Set a spikeinterface variable with the full path of the combinato code; then we can call the scripts using the full path. This makes the calls easy for windows and the idea to set the path of the code is used for other sorters (like ironclus or waveclus because they require to add their path in Matlab).
2. I added all the parameters in options and filters, and now they can be seen and set like the parameters of any other sorter in spikeinterface. 

**How run_combinato works**
1. Saves the recording in a .h5 file (This is one of the cool parts, spikeinterface has to deal with the raw data)
2. Creates a .bat/.sh file with a script that runs css-extract (*) and then css-simple-clustering  (with a flag if the user set a particular type of detection sign )

The code is here: [here](https://github.com/ferchaure/spikesorters/blob/master/spikesorters/combinato/combinato.py) probably the descriptions of the sorter and parameters can be improved... I just copy-pasted and added my broken English on top 

(*) To optimize the running time could be nice to have a way to set the sign if just one type of detection is required.
